File: Daily Challenge/January 2025/Counting Words With a Given Prefix.cpp
class Solution {
  bool isPrefix(const string& word, const string& pref) {
    if (word.size() < pref.size()) return false;
    for (int i = 0; i < pref.size(); ++i) {
      if (word[i] != pref[i]) return false;
    }

    return true;
  }
public:
  int prefixCount(vector<string>& words, string pref) {
    int count = 0;
    for (const string& word : words) {
      if (isPrefix(word, pref)) {
        ++count;
      }
    }

    return count;
  }
};

File: Daily Challenge/January 2025/Counting Words With a Given Prefix.cs
public class Solution {
  private bool isPrefix(string word, string pref) {
    if (word.Length < pref.Length) return false;

    for (int i = 0; i < pref.Length; ++i) {
      if (word[i] != pref[i]) return false;
    }

    return true;
  }
  public int PrefixCount(string[] words, string pref) {
    int count = 0;
    for (int i = 0; i < words.Length; ++i) {
      if (isPrefix(words[i], pref)) count++;
    }

    return count;
  }
}